prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,645 @@
+    return Max_nodeSet
+
+
+def InitKV(clean_data):
+    """Calculate the base_KV for each node based on MIN_VOLT and MAX_VOLT;
+    Save the invalid substations as to be calculated later
+
+    :param pandas.DataFrame clean_data: substation dataframe as returned by :func:`Clean`
+    :return: (*dict*) -- KV_dict, a dict of substation's baseKV value.
+    :return: (*list*) -- to_cal, a list of substations to be calculated using neighbour search.
+    """
+
+    KV_dict = {}
+    to_cal = []
+    Max_Value = 3000
+    Min_Value = 0
+    for index, row in clean_data.iterrows():
+        base_KV = default_base_kv
+        Min_Vol = row["MIN_VOLT"]
+        Max_Vol = row["MAX_VOLT"]
+        if Min_Vol >= Max_Value or Min_Vol <= Min_Value:
+            if Max_Vol < Max_Value and Max_Vol > Min_Value:
+                base_KV = Max_Vol
+            else:
+                to_cal.append((row["LATITUDE"], row["LONGITUDE"]))
+                continue
+        else:
+            if Max_Vol < Max_Value and Max_Vol > Min_Value:
+                base_KV = (Max_Vol + Min_Vol) / 2
+            else:
+                base_KV = Min_Vol
+        KV_dict[(row["LATITUDE"], row["LONGITUDE"])] = base_KV
+    return KV_dict, to_cal
+
+
+# Give a node, find its neighbors in 5 iteration
+
+
+def get_neigbors(g, node, depth=1):
+    """Give a node, find its neighbors in given number depth
+
+    :param networkx.Graph g: Graph generated by :func:`GraphOfNet`
+    :param tuple node: substation coordinate tuple
+    :param int depth: number of hops that we are searching the neighbor substations
+    :return: (*list*) --  a list of closest substations.
+    """
+    output = {}
+    layers = dict(nx.bfs_successors(g, source=node, depth_limit=depth))
+    nodes = [node]
+    for i in range(1, depth + 1):
+        output[i] = []
+        for x in nodes:
+            output[i].extend(layers.get(x, []))
+        nodes = output[i]
+    return output
+
+
+def Cal_KV(N_dict, G, KV_dict, to_cal):
+    """Estimate missing substation's KV data using the neighbors average calculation
+
+    :param dict N_dict: dict of substation's neighbors as returned by :func:`Neighbors`
+    :param networkx.Graph g: Graph generated by :func:`GraphOfNet`
+    :param dict KV_dict: dict of existing baseKV for each substation
+    :param list to_cal: list of the substations to be calculate using average KV
+    :return: (*dict*) --  dict of all substation's baseKV being corrected and calculated.
+    """
+
+    for sub in to_cal:
+        if sub not in N_dict:
+            continue
+        neigh = get_neigbors(G, sub, depth=5)
+        temp_KV = 0
+        num = 0
+        for i in range(1, 6):
+            for nei in neigh[i]:
+                if nei in KV_dict:
+                    temp_KV = temp_KV + KV_dict[nei]
+                    num = num + 1
+            if num > 0:
+                KV_dict[sub] = temp_KV / num
+                continue
+            else:
+                KV_dict[sub] = -999999
```

Comment:
If we assign a voltage of -1 GV to a substation within this loop, could that value get used in calculating voltages for this substations neighbors in future iterations of the loop?

Reply:
this does get used during some estimation in the future iteration and it caused some bus's base_kv being negative as the final results. I have removed this setup in the latest version and cleanup the data.